src/Orleans/Runtime/AsynchAgent.cs

```diff
@@ -250,9 +250,11 @@ public override string ToString()
             return Name;
         }
 
+        internal static SystemStatus CurrentSystemStatus { get; set; } = SystemStatus.Unknown;
+
         private static void LogStatus(Logger log, string msg, params object[] args)
         {
-            if (SystemStatus.Current.Equals(SystemStatus.Creating))
+            if (CurrentSystemStatus.Equals(SystemStatus.Creating))
```

Comment:
This will not behave the same way as before, since you only set it to Creating once and then it stays like that after Creating actually finished. I'm not suggesting to keep them exactly in sync, but if you aren't doing that, you could at least set this when it transitions to Creating and transition off Creating. Also, if you are not keeping them up to date after that, I would just have a boolean flag, like IsStarting